bastion/task/workflow/notion_to_s3.go

```diff
@@ -0,0 +1,357 @@
+func (n *notionToS3Workflow) getImageName(url string) string {
+	// Example url: https://s3.us-west-2.amazonaws.com/secure.notion-static.com/b3b54d2c-a596-4383-8cf9-9a3352631546/rayze_as_cheetah_looking_from_rooftops.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=AKIAT73L2G45EIPT3X45%2F20230322%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20230322T041608Z&X-Amz-Expires=3600&X-Amz-Signature=ed4de5d680af981f7ee5ce12a1366051d7d9314da0db93f249966cfe387212ce&X-Amz-SignedHeaders=host&x-id=GetObject
+	splitStr1 := strings.Split(url, "?")
+	splitStr2 := strings.Split(splitStr1[0], "/")
+	return splitStr2[len(splitStr2)-1]
+}
+
+func (n *notionToS3Workflow) transferImage(url string) (string, error) {
+	imageName, err := n.downloadImage(url)
+	if err != nil {
+		return "", fmt.Errorf("failed to download image: %v", err)
+	}
+
+	output, err := n.s3Client.UploadObject(n.bucket, n.folder+"/"+constant.MediaFolder+"/"+imageName, imageName, true)
+	if err != nil {
+		return "", fmt.Errorf("failed to upload image to S3: %v", err)
+	}
+
+	err = os.Remove(imageName)
+	if err != nil {
+		logger.Errorf("Failed to remove image file %s: %v", imageName, err)
+	}
+
+	return output.Location, nil
+}
+
+func (n *notionToS3Workflow) downloadImage(url string) (string, error) {
+	imageName := n.getImageName(url)
+	imageFile, err := os.Create(imageName)
+	if err != nil {
+		return "", fmt.Errorf("failed to create image file %s: %v", imageName, err)
+	}
+	defer imageFile.Close()
+
+	coverImage, err := http.Get(url)
+	if err != nil {
+		return "", fmt.Errorf("failed to download image: %v", err)
+	}
+	defer coverImage.Body.Close()
+
+	_, err = io.Copy(imageFile, coverImage.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to copy image to the file %s: %v", imageName, err)
+	}
+
+	return imageName, nil
+}
+
+func (n *notionToS3Workflow) createAndUploadContentJson(content interface{}) error {
+	contentJson, err := json.MarshalIndent(content, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal blocks: %v", err)
+	}
+
+	fileHandle, err := os.Create(constant.ContentObject)
+	if err != nil {
+		return fmt.Errorf("failed to create %s", constant.ContentObject)
+	}
+	defer fileHandle.Close()
+
+	_, err = fileHandle.Write(contentJson)
+	if err != nil {
+		return fmt.Errorf("failed to write to file: %v", err)
+	}
+
+	_, err = n.s3Client.UploadObject(n.bucket, n.folder+"/"+constant.ContentObject, constant.ContentObject, false)
+	if err != nil {
+		return fmt.Errorf("failed to upload image to S3: %v", err)
+	}
+
+	err = os.Remove(constant.ContentObject)
+	if err != nil {
+		logger.Errorf("Failed to remove josn file: %v", err)
+	}
+
+	return nil
+}
+
+func (n *notionToS3Workflow) getContent(paragraph *notion.Paragraph) (string, error) {
+	if paragraph == nil || paragraph.RichTexts == nil || len(paragraph.RichTexts) > 1 {
+		return "", fmt.Errorf("invalid paragraph")
+	}
+
+	if len(paragraph.RichTexts) == 0 {
+		return "\n", nil
```

Comment:
why return "\n" by default?  why not just ""? What's the impact?

Reply:
Because when RichTexts is an empty array, it means that line(block) doesn't have anything.  So it indicates a newline.